src/index.ts

```diff
@@ -85,62 +85,62 @@ export default async function GleeAppInitializer () {
     }
   })
 
-  app.on('adapter:connect', (e: EnrichedEvent) => {
+  app.on('adapter:connect', async (e: EnrichedEvent) => {
     logLineWithIcon(':zap:', `Connected to server ${e.serverName}.`, {
       highlightedWords: [e.serverName],
     })
-    runLifecycleEvents('onConnect', {
+    await runLifecycleEvents('onConnect', {
       glee: app,
       serverName: e.serverName,
       connection: e.connection,
     })
   })
   
-  app.on('adapter:reconnect', (e: EnrichedEvent) => {
+  app.on('adapter:reconnect', async (e: EnrichedEvent) => {
     logLineWithIcon('↪', `Reconnected to server ${e.serverName}.`, {
       highlightedWords: [e.serverName],
       iconColor: '#0f0',
     })
-    runLifecycleEvents('onReconnect', {
+    await runLifecycleEvents('onReconnect', {
```

Comment:
I think we should set the `captureRejections` to `true` in `glee.ts` constructor and have a `on('error')` to capture the errors we can handle the rejections here.